Change dhcp to auto in network config

agners opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Breaking change

<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->

## Proposed change
While in IPv4 world auto equals (more or less) to DHCP, this is not
true for IPv6: Besides DHCP address asignment can be done through the
SLAAC mechanism. Let's use "auto" in both cases, which is also more
user friendly.

<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->

## Type of change

<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml

```

## Additional information

<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

agners committed:
Change DHCP to auto in network configuration

While in IPv4 world auto equals (more or less) to DHCP, this is not
true for IPv6: Besides DHCP address asignment can be done through the
SLAAC mechanism.

Jc2k reviewed (dismissed):
:+1: - this has already cropped up a few times when I've been helping users with their Thread devices. There it's really important that they use Auto so that HA can actually use their border routers.  But they are getting configured by the (incorrect) DHCP label, and actually stopping their HA from processing border router route announcements entirely.

bramkragten commented on hassio/src/dialogs/network/dialog-hassio-network.ts:
```diff
@@ -316,7 +316,7 @@ export class DialogHassioNetwork
         <div class="radio-row">
           <ha-formfield
-            .label=${this.supervisor.localize("dialog.network.dhcp")}
+            .label=${this.supervisor.localize("dialog.network.auto")}
```
You should update the actually translation (and key) in `src/en.json`

bramkragten replied:
And then you need to update the translation of the legacy supervisor UI too 😄 
Under `supervisor.dialog.network`

agners committed:
Update translation

agners committed:
Change DHCP to auto in network configuration

While in IPv4 world auto equals (more or less) to DHCP, this is not
true for IPv6: Besides DHCP address asignment can be done through the
SLAAC mechanism.

agners committed:
Add new translation for old Supervisor dialog
